lib/sequelize.js

```diff
@@ -614,24 +614,30 @@ class Sequelize {
         [sql, bindParameters] = this.dialect.Query.formatBindParameters(sql, options.bind, this.options.dialect);
       }
 
+      const checkTransaction = () => {
+        if (options.transaction && options.transaction.finished && !options.completesTransaction) {
```

Comment:
Why `completesTransaction` is added to this check? Is it to let rollback and commit run on already finished transaction?

Reply:
It's because we don't want to actually run this check against a `COMMIT` or `ROLLBACK` query, because `transaction.finished` actually gets set just before the query actually fires off.